drivers/net/bnxt/rte_pmd_bnxt.c

```diff
@@ -450,5 +451,81 @@ int rte_pmd_bnxt_get_vf_tx_drop_count(uint8_t port, uint16_t vf_id, uint64_t *co
 		return -ENOTSUP;
 	}
 
-	return bnxt_hwrm_func_qstats_tx_drop(bp, bp->first_vf_id + vf_id, count);
+	return bnxt_hwrm_func_qstats_tx_drop(bp,
+					     bp->pf.first_vf_id + vf_id, count);
+}
+
+int rte_pmd_bnxt_mac_addr_add(uint8_t port,
+				struct ether_addr *addr,
+				uint16_t vf_id)
+{
+	struct rte_eth_dev *dev;
+	struct rte_eth_dev_info dev_info;
+	struct bnxt *bp;
+	struct bnxt_child_vf_info *vf;
+	struct bnxt_filter_info *filter;
+	struct bnxt_vnic_info vnic;
+	uint16_t *vnic_ids;
+	size_t vnic_id_sz;
+	size_t sz;
+	int rc;
+
+	dev = &rte_eth_devices[port];
+	rte_eth_dev_info_get(port, &dev_info);
+	bp = (struct bnxt *)dev->data->dev_private;
+
+	if (vf_id >= dev_info.max_vfs)
+		return -EINVAL;
+
+	if (!BNXT_PF(bp)) {
+		RTE_LOG(ERR, PMD,
+			"Attempt to config VF %d MAC on non-PF port %d!\n",
+			vf_id, port);
+		return -ENOTSUP;
+	}
+
+	vf = &bp->pf.vf_info[vf_id];
+
+	/* Attach requested MAC address to the new l2_filter */
+	STAILQ_FOREACH(filter, &vf->filter, next) {
+		if (memcmp(addr, filter->l2_addr, ETHER_ADDR_LEN) == 0) {
+			RTE_LOG(ERR, PMD,
+				"Filter already exists for MAC addr\n");
+			return 0;
+		}
+	}
+	filter = bnxt_alloc_filter(bp);
```

Comment:
There is no delete happening in the VFd, like an add.
I will add something in the course of normal cleanup?